### PR TITLE
Add and use AP_TUNING_ENABLED

### PR DESCRIPTION
**Pull request by peterbarker**
```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *      *           *       *                 *      *      *
HerePro             *                                                                     
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                *      *           *       *                 *      *      *
MatekF405                      *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot             *                  *       *                 *      *      *
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      *      *           *       *                 *      *      *
skyviper-v2450                                    *                                       
```

```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       8      *           16      16                -3880  8      16
HerePro             *                                                                     
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                16     *           16      8                 -3216  8      16
MatekF405                      8      *           8       8                 -3208  8      8
Pixhawk1-1M-bdshot             8                  8       8                 -3264  0      0
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      8      *           8       8                 -3224  8      8
skyviper-v2450                                    8                                       
```

### CODE REVIEW

**Review by Hwurzburg — DISMISSED** [review]
I would rather this be AP_TX_TUNING_ENABLED

**Comment by peterbarker** [comment]
> I would rather this be AP_TX_TUNING_ENABLED

Me too.  'though I'd point out it's not the transmitter you're tuning :-)

However, I'm sticking to our convention with the define name here (I've asked others to do the same before, so I'm not making an exception for me!).

File: Tools/scripts/build_options.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def __init__(self,
     Feature('Plane', 'DEEPSTALL', 'HAL_LANDING_DEEPSTALL_ENABLED', 'Enable Deepstall Landing', 0, None),
     Feature('Plane', 'QAUTOTUNE', 'QAUTOTUNE_ENABLED', 'Enable QuadPlane Autotune mode', 0, "QUADPLANE"),
     Feature('Plane', 'PLANE_BLACKBOX', 'AP_PLANE_BLACKBOX_LOGGING', 'Enable blackbox logging', 0, None),
+    Feature('Plane', 'AP_TUNING', 'AP_TUNING_ENABLED', 'Enable two-switch parameter tuning', 0, None),
```
Comment (Hwurzburg) [inline Tools/scripts/build_options.py:187]: then change this so as to not confuse users
```suggestion
    Feature('Plane', 'Transmitter based tuning', 'AP_TUNING_ENABLED', 'Enable TX based tuning parameter adjustments', 0, None),
```

Comment (peterbarker) [reply Tools/scripts/build_options.py]: Done, thanks.

**Comment by peterbarker** [comment]
@Hwurzburg discussed on call, tridge isn't averse to renaming the library to something like `AP_Transmitter_Tuning`.  I'm still a little uncomfortable with that name.

**Pull request merged**